streamlit-app.py

```diff
@@ -59,46 +61,98 @@ def extract_model_names(models_info):
         # models_dicts = [asdict(model) for model in models_info]
         # print(models_dicts)
         return tuple(models.model for models in models_info)
-
-def create_vector_db(file_upload) -> Chroma:
+    
+def read_files_and_split(file_uploads: list[UploadedFile]) -> list:
     """
-    Create a vector database from an uploaded PDF file.
+    Read multiple files, extract their text, and split them into chunks.
 
     Args:
-        file_upload (st.UploadedFile): Streamlit file upload object containing the PDF.
+        file_uploads (list[st.UploadedFile]): List of Streamlit file upload objects containing PDFs.
 
     Returns:
-        Chroma: A vector store containing the processed document chunks.
+        list: A list of document chunks.
     """
-    logger.info(f"Creating vector DB from file upload: {file_upload.name}")
+    logger.info("Reading files and splitting into chunks")
     temp_dir = tempfile.mkdtemp()
+    all_chunks = []
+
+    try:
+        for file_upload in file_uploads:
+            logger.info(f"Processing file: {file_upload.name}")
+            # Save file temporarily
+            path = os.path.join(temp_dir, file_upload.name)
+            with open(path, "wb") as f:
+                f.write(file_upload.getvalue())
+                logger.info(f"File saved to temporary path: {path}")
+                
+                if path.endswith(".pdf"):
+                    # Load the PDF
+                    loader = PyPDFLoader(path)
+                elif path.endswith(".docx"):
+                    loader = UnstructuredWordDocumentLoader(path)
+                data = loader.load()
+                logger.info(f"Loaded {len(data)} documents from {file_upload.name}")
+
+                # Split the text into chunks
+                text_splitter = RecursiveCharacterTextSplitter(chunk_size=7500, chunk_overlap=100)
+                chunks = text_splitter.split_documents(data)
+                all_chunks.extend(chunks)  # Accumulate chunks
+
+        logger.info(f"Total chunks created: {len(all_chunks)}")
+        return all_chunks
+
+    finally:
+        # Clean up temporary files
+        shutil.rmtree(temp_dir)
+        logger.info(f"Temporary directory {temp_dir} removed")
+
+
+def create_vector_store(chunks: list, collection_name: str = "myRAG", persist_directory: str = "./app") -> Chroma:
```

Comment:
we can make .env file and each one store his own environment values into that file such as the local directories and any custom values and pull the values from it to not have conflicts related to the directories names because i think that persist_directory on my branch was ./chromdb so it is better to not hardcode it. So, i will make that file and try to store any local configuration into it and that will be added to the .gitignore